pkgs/top-level/splice.nix

```diff
@@ -37,26 +37,25 @@ let
       inherit name;
       value = let
         defaultValue = mash.${name};
-        buildValue = buildPkgs.${name};
-        runValue = runPkgs.${name};
+        # `or {}` is for the non-derivation attsert splicing case, where `{}` is the identity.
+        buildValue = buildPkgs.${name} or {};
+        runValue = runPkgs.${name} or {};
         augmentedValue = defaultValue
           // (lib.optionalAttrs (buildPkgs ? ${name}) { nativeDrv = buildValue; })
           // (lib.optionalAttrs (runPkgs ? ${name}) { crossDrv = runValue; });
-        # Get the set of outputs of a derivation
+        # Get the set of outputs of a derivation. If one derivation fails to
+        # evaluate we don't want to diverge the entire splice, so we fall back
+        # on {}
         tryGetOutputs = value0: let
-          eval = builtins.tryEval value0;
-        in getOutputs (if eval.success then eval.value else {});
+          inherit (builtins.tryEval value0) success value;
+        in tryGetOutputs (lib.optionalAttrs success value)
         getOutputs = value: lib.genAttrs
           (value.outputs or (lib.optional (value ? out) "out"))
           (output: value.${output});
       in
-        # Certain *Cross derivations will fail assertions, but we need their
```

Comment:
This old branch was silly. If `defaultValue` diverges, `augmentedValue` will too.